state_machine/applications/system-check/tests/coil_test.py

```diff
@@ -0,0 +1,146 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Torque Driver Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from lib import pycubed as cubesat
+from ulab import numpy
+
+# voltage level constants; set between -1 and 1
+v1 = 0.25
+v2 = 0.5
+v3 = 0.75
+
+# find projected voltage: vlevel * max voltage
+projected_voltage1 = v1 * 5.06
+projected_voltage2 = v2 * 5.06
+projected_voltage3 = v3 * 5.06
+
+# alias norm
+norm = numpy.linalg.norm
+
+
+def test_voltage_levels(coil_index):
+    """
+    Ask the user to type y/any key to start/cancel the test
+    If y, collect IMU magnetometer data for different voltage levels,
+    and return if the magnetometer readings are increasing as voltage
+    levels are increasing.
+    If any other key, return  None
+    """
+
+    # wait times
+    driver_time = 1
+
+    # set up the test
+    print(f"""Testing Coil Driver {coil_index} for the following voltage
+levels: {projected_voltage1} V, {projected_voltage2} V, {projected_voltage3} V.""")
+
+    # conduct the test
+    start_test = input("Type Y to start the test, any key to cancel: ")
+    if start_test.lower() != "y":
+        return None, None
+
+    # else, test each voltage level
+    cubesat.coildriver_vout(coil_index, projected_voltage1)
+    time.sleep(driver_time)
+    mag_reading1 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage2)
+    time.sleep(driver_time)
+    mag_reading2 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage3)
+    time.sleep(driver_time)
+    mag_reading3 = cubesat.magnetic()
+
+    # calculate total magnetic reading
+    print("Data Collection Complete")
+    mag_total1 = norm(mag_reading1)
+    mag_total2 = norm(mag_reading2)
+    mag_total3 = norm(mag_reading3)
+
+    # if magnetometer readings are increasing over time, return true
+    # for reference, an average fridge magnet has a force of around 1000 µT
```

Comment:
This is probably a question for @NeilKhera/@jbwillis but, does this sound like a reasonable strength?
On [wikepedia](https://en.wikipedia.org/wiki/Tesla_(unit)#Examples) I found that being 5cm away from a vacuum cleaner  is about 40µT.

Now from my sketchy memory of E&M, I believe if you have an infinitely long current conducting wire, magnetic field strength is inversely proportional to distance.  So even if we were 1cm away from the vacuum cleaner it would be about 200µT. 

This is all super rough, but it seems a little high.

Reply:
This number is based off of a google search, so it could very well be incorrect. I believe the best way to set a threshold is to actually pick a "test" magnet and redefine the bounds based on the readings we get.

I also noticed something weird - the average reading I'm getting is around 180 µT (3.5 times the Earth's magnetic field in San Jose). I'm not sure if the magnetometer is broken, or if there's actually that much of an induced magnetic field from my computer, wires, and the board.

Reply:
Yeah, I think this is probably something worth testing.

Here's my estimate using this calculator http://hyperphysics.phy-astr.gsu.edu/hbase/magnetic/curloo.html. 
We have 18 loops * 4 layers * 2 boards = 144 loops.
So N = 150 (approximately)
As I recall, each board is approximately 100 ohms, so 200 ohms per coil pair.
At 4 volts, that's
I = 20mA.
We multiply I by the number of loops so the effective
I = 3A.
The boards are approximately 5cm wide, so 
R = 0.025m.

That's giving ~750 uT at the center of the coils, and ~25uT at the approximate location of the magnetometer. So yes, that `100` number is too large. I'd say let's drop it to `1uT`, just to give a small margin it has to beat. 

Fair point about the earths magnetic field. To handle that, a reading at 0V should be made and (vector) subtracted off of the other readings before taking their norm.

![image](https://user-images.githubusercontent.com/29437542/183506523-15954579-6a5f-453e-b9bb-e2ffb9083a41.png)

Reply:
I'm glad you've looked into it, if you've already done some research like that I think it works great as a placeholder value. 

Also, honestly for being indoors that sounds totally normal. Your phone does have a magnetometer though so you could compare to that.

Thinking some more, why do we just move forward with this after you do two things:
1. Leave a comment saying that this 1000μT is a placeholder.
2. Change the comment from "has a **force** of around..." to "has a **strength** of around...". Just because tesla's aren't a measure of force.

Preemptively approved.

Reply:
I'd actually like to make the change to have the background (0Volt) measurement subtracted off. Then, I agree that the number is more of a placeholder at this point

Reply:
Oh, I didn't see your comment @jbwillis, before I left mine. I think that makes a lot of sense, although the `1μT` threshold seems a bit low, I think sensor noise should be more than enough to set that off. What do you think?

Reply:
I honestly have no idea what the noise level is. Maybe 10 would be more appropriate.

Reply:
I think 10 is a better threshold to beat than 1 - sensor noise definitely would account for 1-2 µT. I'll also add in a 0V measurement and subtract it from the other readings before taking the norm; this seems like a good way to ensure the numbers are more trustworthy.

Thank you for looking into it!